src/main/java/com/gluonhq/substrate/target/IosTargetConfiguration.java

```diff
@@ -110,23 +111,31 @@ public IosTargetConfiguration(ProcessPaths paths, InternalProjectConfiguration c
 
     @Override
     List<String> getTargetSpecificCCompileFlags() {
-        return Arrays.asList("-xobjective-c",
+        List<String> flags = new ArrayList<>(Arrays.asList("-xobjective-c",
                 "-arch", getTargetArch(),
                 "-mios-version-min=11.0",
-                "-I"+projectConfiguration.getGraalPath().resolve("include").toString(),
-                "-I"+projectConfiguration.getGraalPath().resolve("include").resolve("darwin").toString(),
-                "-isysroot", getSysroot());
+                "-I" + projectConfiguration.getGraalPath().resolve("include").toString(),
+                "-I" + projectConfiguration.getGraalPath().resolve("include").resolve("darwin").toString(),
+                "-isysroot", getSysroot()));
+        if (isSimulator()) {
+            flags.add("-DGVM_IOS_SIM");
+        }
+        return flags;
     }
 
     @Override
     List<String> getTargetSpecificAOTCompileFlags() throws IOException {
-        return Arrays.asList("-H:CompilerBackend=" + Constants.BACKEND_LLVM,
-                "-H:-SpawnIsolates",
+        List<String> flags = new ArrayList<>(Arrays.asList(
                 "-H:PageSize=16384",
                 "-Dsvm.targetName=iOS",
                 "-Dsvm.targetArch=" + getTargetArch(),
                 "-H:+UseCAPCache",
-                "-H:CAPCacheDir=" + getCapCacheDir().toAbsolutePath().toString());
+                "-H:CAPCacheDir=" + getCapCacheDir().toAbsolutePath().toString()));
+        if (!isSimulator()) {
```

Comment:
Interesting, does this imply that the reason of using LLVM backend is because of the aarch64 arch, not because of the iOS OS?

Reply:
It is not the arch: Android for instance has aarch64 too, but doesn't need LLVM (we did use it some time ago though). We use the LLVM backend for iOS/arm64 to create LLVM IR bitcode.
For the simulator (x86_64) this is not needed.